new/node_state_separation.md

```diff
@@ -0,0 +1,287 @@
+* 403 Forbidden.  The user which made the request is not authorized to perform the action.
+* 404 Not found.  The requested object does not exist.
+* 413 Request entity too large. A request may not be larger than 1000000 bytes.
+
+##### `POST /nodes/:name/configuration`
+
+This method will not be implemented, use `POST /nodes` to create a new node object.
+
+##### `DELETE /nodes/:name/configuration`
+
+This method will not be implemented, partial clearing of data will not be supported.  Use `DELETE /nodes/:name`.
+
+#### `/nodes/:name/runtime`
+
+This endpoint is for storing the runtime data (default attributes, override attributes, automatic attributes)
+
+##### `GET /nodes/:name/runtime`
+
+The `GET` method is used to return the runtime details of the node as JSON.
+
+This method has no parameters.
+
+The response will be similar to:
+
+```
+{
+  "name": "node_name",
+  "default": {
+    [...]
+  },
+  "override": {
+    [...]
+  },
+  "automatic": {
+    [...]
+  }
+}
+```
+
+Response Codes:
+
+* 200 OK.  The request was successful
+* 401 Unauthorized.  The user or client who made the request could not be authenticated. Verify the user/client name, and that the correct key was used to sign the request.
+* 403 Forbidden.  The user which made the request is not authorized to perform the action.
+* 404 Not found.  The requested object does not exist.
+
+##### `PUT /nodes/:name/runtime`
+
+The `PUT` method is used to update the runtime data on an already created node.
+
+This method has no parameters.
+
+The request body will be similar to:
+
+```
+{
+  "name": "node_name",
+  "default": {
+    [...]
+  },
+  "override": {
+    [...]
+  },
+  "automatic": {
+    [...]
+  }
+}
+```
+
+The response will return the updated configuration data in the same format as the request.
+
+Response Codes:
+
+* 200 OK.  The request was successful
+* 401 Unauthorized.  The user or client who made the request could not be authenticated. Verify the user/client name, and that the correct key was used to sign the request.
+* 403 Forbidden.  The user which made the request is not authorized to perform the action.
+* 404 Not found.  The requested object does not exist.
+* 413 Request entity too large. A request may not be larger than 1000000 bytes.
+
+##### `POST /nodes/:name/runtime`
+
+This method will not be implemented, use `POST /nodes` to create a new node object.
+
+##### `DELETE /nodes/:name/runtime`
+
+This method will not be implemented, partial clearing of data will not be supported.  Use `DELETE /nodes/:name`.
+
+#### `/nodes/:name`
+
+This endpoint is unchanged, and is only included here for context.
+
+##### `GET /nodes/:name`
+
+This method will return the combined configuration and runtime data and its semantics will be unchanged.
+
+##### `PUT /nodes/:name`
+
+This method will create a new node with the combined configuration and runtime data and its semantics will be unchanged.
```

Comment:
All of this needs to be updated with the discussion on ACLs.  I'm not planning on documenting how the API is used in the API documentation since that may change, the dirty config stuff is entirely client-side so wouldn't be documented here.